### DISCUSSION TITLE
Deprecated package, what package should I use?

### DISCUSSION THREAD

**Discussion by TheVeryStarlk** [dotnet/aspnetcore#49783] [Q&A] [answered]
The [Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets](https://www.nuget.org/packages/Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets) package is deprecated and I would like to use it's `SocketTransport` implementations. Is there a package I should use that replaces the deprecated one?

**Comment by LeaFrock** [comment, marked as answer]
It should be already included in aspnetcore sdk(3.0+), instead of exposing as a single package. 

If you're using aspnetcore, it's recommended upgrading the framework to .NET 6+ and use it directly.  If you aren't, make sure your project refer to the sdk. See [this doc](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-7.0&tabs=visual-studio#use-the-aspnet-core-shared-framework) for details.

**Reply by TheVeryStarlk** [reply]
Hi, thanks for the reply! The mentioned package `Microsoft.AspNetCore.App` seems to be deprecated too. https://www.nuget.org/packages/Microsoft.AspNetCore.App

**Reply by LeaFrock** [reply]
Notice that the doc says `FrameworkReference` rather than `PackageReference`. Take a try first if you have any concerns.

**Reply by TheVeryStarlk** [reply]
You're right, I did not notice that it uses `FrameworkReference`. Thanks!
